### PR TITLE
update prompts and command line arguments for office-generator project

### PR DESCRIPTION
**Pull request by TCourtneyOwen**
- Remove prompting for folder name - will now default to project name. If user wants to specify a folder name different than the project name, they can use the new 'output" param
- Remove prompting for manifest-only. User can specify manifest-only now that it is included as one of the project types (i.e. it's now listed on the prompts with react., jquery and angular). 
- Remove prompt for TypeScript.  We now default to Typescript. If the user wants to use JS instead, they can use the 'js' param from the command line
- Change "framework" to "projectType" throughout
- Add some additional output logging to command line
- Minor update to readme.md to cite new 'output' param
- Update test classes to accommodate changes to command prompts
- Add new output test case to new-project.ts

### CODE REVIEW

File: src/app/index.ts [hunk 1]
```diff
@@ -101,147 +98,89 @@ module.exports = yo.extend({
...
-      let startForFramework = (new Date()).getTime();
-      let askForFramework = [
+      /** technology used to create the addin (jquery / angular / etc) */
+      let startForProjectType = (new Date()).getTime();
+      let askForProjectType = [
         {
-          name: 'framework',
-          message: 'Choose a framework:',
+          name: 'projectType',
+          message: 'Choose a project-type:',
```
Comment (casieber) [inline src/app/index.ts]: nit - Since this is just an English prompt can we change "project-type" to "project type"?

File: src/app/index.ts [hunk 2]
```diff
@@ -101,147 +98,89 @@ module.exports = yo.extend({
...
           choices: tsTemplates.map(template => ({ name: _.capitalize(template), value: template })),
-          when: (this.project.framework == null) && this.project.ts && !this.options.js && !answerForManifestOnly.isManifestOnly
+          when: (this.project.projectType == null || !this._isValidInput(this.options.projectType, tsTemplates, false /* isHostParam */)) && this.project.ts && !this.options.js
+                && !this.project.isManifestOnly
         },
         {
-          name: 'framework',
-          message: 'Choose a framework:',
+          name: 'projectType',
+          message: 'Choose a project-type:',
```
Comment (casieber) [inline src/app/index.ts]: nit - Same as above

File: src/app/index.ts [hunk 3]
```diff
@@ -288,20 +224,20 @@ module.exports = yo.extend({
         /** Copy the manifest */
         this.fs.copyTpl(this.templatePath(`manifest/${this.project.hostInternalName}.xml`), this.destinationPath(`${this.project.projectInternalName}-manifest.xml`), templateFills);
 
-        if (this.project.framework === 'manifest-only') {
-          this.fs.copyTpl(this.templatePath(`manifest-only/**`), this.destinationPath(), templateFills);
+        if (this.project.projectType === manifestOnly) {
+          this.fs.copyTpl(this.templatePath(manifestOnly.concat(`/**`)), this.destinationPath(), templateFills);
```
Comment (casieber) [inline src/app/index.ts]: Use a template literal here for consistency with the others: `` `${manifestOnly}/**` ``

File: src/app/index.ts [hunk 4]
```diff
@@ -334,8 +267,8 @@ module.exports = yo.extend({
         this.installDependencies({
           npm: true,
           bower: false,
-          callback: this._exitProcess.bind(this)
-        });
+          callback: this._postInstallHints.bind(this)
+        });
```
Comment (casieber) [inline src/app/index.ts]: nit - Extra whitespace at end of line

File: src/test/manifest-only-project.ts [hunk 5]
```diff
@@ -72,10 +82,9 @@ describe('manifest-only project - answers & args - jquery & typescript', () => {
   describe('argument: name', () => {
     before((done) => {
       answers.host = 'excel';
-      answers.isManifestOnly = true;
       argument[0] = projectEscapedName;
 
-      helpers.run(path.join(__dirname, '../app')).withArguments(argument).withPrompts(answers).on('end', done);
+      helpers.run(path.join(__dirname, '../app')).withArguments(argument).withPrompts(answers).on('end', done)
```
Comment (casieber) [inline src/test/manifest-only-project.ts]: nit - missing semicolon

File: src/test/manifest-only-project.ts [hunk 6]
```diff
@@ -106,7 +112,7 @@ describe('manifest-only project - answers & args - jquery & typescript', () => {
       argument[0] = projectEscapedName;
       argument[1] = 'excel';
 
-      helpers.run(path.join(__dirname, '../app')).withArguments(argument).withPrompts(answers).on('end', done);
+      helpers.run(path.join(__dirname, '../app')).withArguments(argument).withPrompts(answers).on('end', done)
```
Comment (casieber) [inline src/test/manifest-only-project.ts]: nit - missing semicolon

File: src/app/index.ts [hunk 7]
```diff
@@ -32,7 +34,7 @@ module.exports = yo.extend({
 
     this.argument('name', { type: String, required: false });
     this.argument('host', { type: String, required: false });
-    this.argument('framework', { type: String, required: false });
+    this.argument('projectType', { type: String, required: false });
```
[the remark this answers is not in the document]
Comment (TCourtneyOwen) [reply src/app/index.ts]: Yes, that's my understanding based on discussions I have had with @akrantz

File: src/app/index.ts (hunk 1 above)
Comment (TCourtneyOwen) [reply src/app/index.ts]: fixed locally

File: src/app/index.ts (hunk 2 above)
Comment (TCourtneyOwen) [reply src/app/index.ts]: fixed locally

File: src/test/manifest-only-project.ts (hunk 5 above)
Comment (TCourtneyOwen) [reply src/test/manifest-only-project.ts]: fixed locally

File: src/test/manifest-only-project.ts (hunk 6 above)
Comment (TCourtneyOwen) [reply src/test/manifest-only-project.ts]: fixed locally

**Comment by akrantz** [comment]
Here's the current help syntax. 
```
c:\dev\office\addin\generator-office-test\test1>yo office --help
Usage:
  yo office:app [options] [<name>] [<host>] [<projectType>]

Options:
  -h,   --help          # Print the generator's options and usage
        --skip-cache    # Do not remember prompt answers                      Default: false
        --skip-install  # Do not automatically install dependencies           Default: false
        --js            # Use JavaScript templates instead of TypeScript.
        --output        # Project folder name if different from project name

Arguments:
  name           Type: String  Required: false
  host           Type: String  Required: false
  projectType    Type: String  Required: false
```
I think we can make some improvements.

1. put projectType first, then name, then host.
2. The arguments section is useless with Type and Required. Can we actually put useful info for the arguments? For example, can we made a table of project type name and description?

angular    | Office Add-in using Angular framework
jquery      | Office Add-in using Jquery framework
react        |  Office Add-in using React framework
manifest  | Office Add-in manifest

3. Add -o for --output.
4. For --js, change help text to "Project uses JavaScript instead of TypeScript"

File: readme.md [hunk 8]
```diff
@@ -58,8 +58,8 @@ The Microsoft Office client application that can host the add-in. The supported
   - Type: String
   - Optional
 
-#### `framework`
-Framework to use for the project. The supported arguments include JQuery (`jquery`), Angular (`angular`), and React (`react`). You can also use Manifest Only (`manifest-only`) which will create only the `manifest.xml` for an Office Add-in.
+#### `projectType`
+Framework to use for the project. The supported arguments include JQuery (`jquery`), Angular (`angular`), and React (`react`). You can also use Manifest-Only (`manifest-only`) which will create only the `manifest.xml` for an Office Add-in.
```
[the remark this answers is not in the document]
Comment (akrantz) [reply readme.md]: Well, we made need to have the info in both places. I'd like to have the command line --help option be more useful, but we might need to also keep the readme info in sync.

File: src/app/index.ts [hunk 9]
```diff
@@ -23,6 +23,8 @@ delete insight.context.tags['ai.device.osVersion'];
 delete insight.context.tags['ai.device.osArchitecture'];
 delete insight.context.tags['ai.device.osPlatform'];
 
+const manifestOnly = 'manifest-only';
```
Comment (akrantz) [inline src/app/index.ts]: Let's simplify the string "manifest-only" to just be "manifest" as the keyword for the project type.

File: src/app/index.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (akrantz) [reply src/app/index.ts]: Yes, we need the flexibility to add different types of projects in the future beyond a single type of add-in with a choice of framework. 

Keyword | Description
----------|------------------------
angular | Office Add-in using Angular 
jquery | Office Add-in using JQuery 
react | Office Add-in using React 
manifest | Office Add-in manifest
excel-functions | Excel Custom Functions
sdx | Office SDX
etc.

**Comment by akrantz** [comment]
@TCourtneyOwen We should clarify how things work if --output option and name argument are specified or not.  I created an empty test1 folder and then ran "yo office test1 excel react" and it made a test1 subfolder. We might want to check whether a folder is empty and use that as well so you don't create a project within an existing dir full of stuff but handle the cases elegantly.

File: src/app/index.ts (hunk 9 above)
Comment (TCourtneyOwen) [reply src/app/index.ts]: fixed locally

File: src/app/index.ts [hunk 10]
```diff
@@ -62,24 +70,13 @@ module.exports = yo.extend({
   prompting: async function () {
     try {
       let jsTemplates = getDirectories(this.templatePath('js'));
+      jsTemplates.push(manifestOnly);
       let tsTemplates = getDirectories(this.templatePath('ts'));
+      tsTemplates.push(manifestOnly);
       let manifests = getFiles(this.templatePath('manifest')).map(manifest => _.capitalize(manifest.replace('.xml', '')));
```
[the remark this answers is not in the document]
Comment (TCourtneyOwen) [reply src/app/index.ts]: fixed locally

**Comment by TCourtneyOwen** [comment]
I am going to close this PR, as it's gotten quite stale

**Pull request closed without merging**